src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/servicebus/ServiceBusHelper.java

```diff
@@ -69,7 +75,17 @@ Message mapToBusMessage(Msg msg) {
         busMessage.setMessageId(msg.getMsgId());
         busMessage.setMessageBody(getMsgBodyInBytes(msg));
         busMessage.setLabel(msg.getLabel());
-
+        busMessage.setProperties(new HashMap<String,String>());
+        RequestTelemetryContext context = ThreadContext.getRequestTelemetryContext();
+        String parentId;
+        if (context != null
+            && context.getHttpRequestTelemetry() != null
+            && context.getHttpRequestTelemetry().getId() != null) {
+            parentId = context.getHttpRequestTelemetry().getId();
+        } else {
+            parentId = LocalStringsUtils.generateRandomIntegerId();
+        }
+        busMessage.getProperties().put(PARENT_ID, parentId);
```

Comment:
how and where we are going to use this id ?

Reply:
reader will get this and put to telemetry request. and if reader process produce any queue message we can use this id as parent id.

Reply:
Message producers will try to keep this id if it is a result of another request. we can introduce another id like span-id(operation-id) it will change with each process.
for example if this envelope has payment it will create a message in payment queue with same parent-id but different  span-id (operation-id)?